Modules/VAMI/VAMI.psm1

```diff
@@ -655,38 +655,46 @@ Function Get-VAMIUser {
     $userAPI = Get-VAMIServiceAPI -NameFilter "accounts"
     $UserResults = @()
 
-    if (($Name -ne "") -and ($null -ne $Name)) {
-        try {
-            $Users = $UserAPI.get($name)
-        } catch {
-            Write-Error $Error[0].exception.Message
-        }
-    } else {
+    # Get a list of users
+    try {
         $Users = $UserAPI.list()
+    } catch {
+        write-error $_
+    }
+
+    # Apply filtering if Name input is provided
+    if ($Name -ne '' -AND $Name -ne $null) {
+        # For 6.5 API, the username is part of the list returnset; for 6.7/7.x API the value from the list is the username. Because of this we will use an OR filter to account for either case.
+        $Users = $Users | Where-Object {$_.username -eq $name -OR $_.value -eq $name}
     }
+
     if ($Users.status) {
+        # This is for 6.5 API, which has a status property; in newer API response there is an enabled property with values of True/False
         foreach ($User in $Users) {
             $UserString = [pscustomobject] @{
                 User = $User.username
                 Name = $User.fullname
                 Email = $User.email
-                Status = $User.status

```

Comment:
Removing the status property breaks backward compatibility.  It's ok to add Enabled property, but you should not remove the status property if you don't want to break the backward compatilibity.

Reply:
Thanks for the tip, I tried to implement this but ran into a couple issues.  First, I don't seem to be able to pass an argument into the scriptblock, so the value 'Enabled' is returned for all users, even if the account is disabled.  Also, the warning text appears for each user in $users instead of just on the first use of the function like other cmdlets I've seen.  I have an idea on how to overcome both of these issues using a global variable, but that doesn't feel like the right answer. 
 I've done a bit of searching but don't understand how to cleanly overcome either of these issues.  Any suggestions on how to handle either of them?

Reply:
It is strange that you cannot pass an argument to the script block. I tried it like this and it seems to work fine:
$UserString = [pscustomobject] @{
    User = 'kamen'
    Name = 'KAmen nikolov'
    Email = 'knikolov@vmware.com'
    Enabled = $true    
}

$myvalue = "Disabled"

$memberParam = @{
    MemberType = "ScriptProperty"
    InputObject = $UserString
    Name = "Status"
    Value = {
        Write-Warning "'Status' property is deprecated. Use 'Enabled' instead."
        return $myvalue
    }
}
Add-Member @memberParam

I cannot figure out a way to not display the warning more than once without some kind of global variable. We have similar static variable in PowerCLI to keep the information about the warnings that have already been displayed. We also usually remove the property from the default display of the type, but here there is no default display, so I would say that a global variable should be ok.

Reply:
I was able to recreate the problem I was seeing... it wasn't that I couldn't pass an argument to the script block, its that the ScriptProperty that is added has the value of the last element in $users, not the correct value for each user.  For example, if I'm using the same foreach loop present in the module like this:

```
$UserResults = @()

$users = @"
User,Name,Email,Enabled
Kamen,Kamen Nikolov,user1@example.com,Enabled
Brian,Brian Wuchner,user2@example.com,Disabled
John,John Doe,user3@example.com,OddValue
"@ | ConvertFrom-Csv

foreach($user in $users) {
    $UserString = [pscustomobject] @{
        User = $user.user
        Name = $user.name
        Email = $user.email
        Enabled = $user.Enabled
    }

    $memberParam = @{
        MemberType = "ScriptProperty"
        InputObject = $UserString
        Name = "Status"
        Value = {
            if ( !$global:vamiDeprecatedPropertyWarningSeen.UserStatus ) { 
                $global:vamiDeprecatedPropertyWarningSeen += @{'UserStatus' = $true }
                write-warning "The status property is deprecated. Use Enabled instead."
            }
            return $user.Enabled
        }
    } # end memberParam
    Add-Member @memberParam

    $UserResults += $UserString
} # end foreach

$UserResults
```

The output is displayed like this:
```
User  Name          Email             Enabled  Status
----  ----          -----             -------  ------
Kamen Kamen Nikolov user1@example.com Enabled  OddValue
Brian Brian Wuchner user2@example.com Disabled OddValue
John  John Doe      user3@example.com OddValue OddValue
```
And as you can see the status for all users is the `OddValue` from John Doe, as he is the last element in $users.

What I was hoping for was that the properties returned for both 6.5 and 6.7+ functions would be the same, which is why I haphazardly deprecated the status property for the 6.5 response and standardized on enabled only.  I'm thinking that an alternative solution would be to just provide both status & enabled properties for both 6.5 and 6.7+ responses.  I have just added that in and submitted a new commit for review.